src/intrinsic/annotations.rs

```diff
@@ -140,6 +145,17 @@ impl AnnotationUnsoundnessError {
                         "The calculus of the called procedure must match the calculus of the calling procedure.",
                     ))
             }
+            AnnotationUnsoundnessError::UnsoundRecursion{direction, span, calculus_name} => {
+                Diagnostic::new(ReportKind::Error, span)
+                    .with_message(format!(
+                        "In {}s, The '{}' calculus does not support calls that may lead to recursion.",
```

Comment:
"The" should not be capitalized.

Maybe reformulate the error message to be more precise. E.g. "Potentially recursive calls are not allowed in proc [NAME] with the 'wp' calculus.". And you could add a hint like, "Procedure calls make use of Park induction, which is not sound in this setting in general."